### PR TITLE
Optimized functions using numba

### PR DESCRIPTION
**Pull request by shreyansjain012**
Optimizes various functions using numba
## Description

**calculate_transition_probabilities()**
Currently the `macro_atom.pyx` is using cython, this file needs to be modified to `macro_atom.py` and to use numba instead, to speed up the code. Also from `tardis/plasma/setup_package.py` the function `get_extension()` is needed to be removed because pytests automatically searches for `get_extenstions()` in every `setup_package.py` file to create an extension of `*.pyx` files to be imported to other files 
   - Converted `tardis/plasma/properties/util/macro_atom.pyx` into 'tardis/plasma/properties/util/macro_atom.py`
   - `tardis/plasma/properties/util/macro_atom.py`: Used `numba.njit` and `numba.prange` to optimize the function `calculate_transition_probabilities()` 
   - `tardis/plasma/setup_package.py` removed `get_extensions()`

**calculate()**
   - `tardis/plasma/properties/general.py`: In Class GElectron optimized function `calculate()` using numba

**calculate()**
   - `tardis/plasma/properties/general.py`: In Class ElectronTemperature optimized function `calculate()` using numba

**calculate()**
   - `tardis/plasma/properties/continuum_processes.py`: Added new function `_calculate()` which optimizes `calculate()` in Class SpontRecombRateCoeff 
## Motivation and Context

Cython uses lots of boilerplate code which makes code unreadable, numba avoids that and also gives similar computation time as Cython. Other functions which can be optimized using numba, should be optimized in order to give fast computation

## How Has This Been Tested?

```

ncalls	tottime	 percall  cumtime  percall   filename:lineno(function)
		#Without numba
21	0.238	 0.011	  0.238	   0.011     {tardis.plasma.properties.util.macro_atom.calculate_transition_probabilities}

21    	0.001    0.000    0.010    0.000     /home/shreyans/Projects/tardis/tardis/plasma/properties/general.py:44(calculate)
		
		#With numba
21 	0.168	 0.008	  0.168	   0.008     /home/shreyans/Projects/tardis/tardis/plasma/properties/util/macro_atom.py:12(calculate_transition_probabilites)

21    	0.000    0.000    0.000    0.000     /home/shreyans/Projects/tardis/tardis/plasma/properties/general.py:46(calculate)
```
**Although 2 functions are still not profiled yet, still committed to show my work. Will add their performance soon**

**Noticed that some functions which can be optimized using numba are not getting called when following quickstart guide, after searching a while I think these functions are called while using GUI but since GUI is not working at the moment I couldn't profile them**

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by shreyansjain012** [comment]
@wkerzendorf @harpolea please review it again

**Comment by wkerzendorf** [comment]
looks nice!!!

**Comment by shreyansjain012** [comment]
Thank You

**Comment by wkerzendorf** [comment]
figure out yourself how you can best show your skills next.

**Comment by shreyansjain012** [comment]
After using `prange` instead of `range` in for loops
image

File: tardis/plasma/setup_package.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
-from setuptools import Extension
+#from setuptools import Extension
```
Comment (wkerzendorf) [inline tardis/plasma/setup_package.py]: I think you can remove that.

**Comment by wkerzendorf** [comment]
> After using `prange` instead of `range` in for loops
> image

how much was it before?

File: tardis/plasma/setup_package.py (hunk 1 above)
Comment (shreyansjain012) [reply tardis/plasma/setup_package.py]: Sure.

**Comment by shreyansjain012** [comment]
> > After using `prange` instead of `range` in for loops
> > image
> 
> how much was it before?

21	0.238	 0.011	  0.238	   0.011     {tardis.plasma.properties.util.macro_atom.calculate_transition_probabilities}

**Pull request closed without merging**